In [1]:
import pandas as pd

In [12]:
df = pd.read_csv("https://raw.githubusercontent.com/Projetar-Solucoes/materiais-trilha/main/semana-04/solicitacoes_semana4.csv")
print(df.shape)

(246, 11)


resultado: (216, 11)

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246 entries, 0 to 245
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   protocolo        246 non-null    object 
 1   data_abertura    246 non-null    object 
 2   solicitante      246 non-null    object 
 3   setor            236 non-null    object 
 4   categoria        246 non-null    object 
 5   assunto          246 non-null    object 
 6   prioridade       232 non-null    object 
 7   status           246 non-null    object 
 8   data_fechamento  135 non-null    object 
 9   avaliacao        124 non-null    float64
 10  unidade          246 non-null    object 
dtypes: float64(1), object(10)
memory usage: 21.3+ KB


reesultado:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246 entries, 0 to 245
Data columns (total 11 columns):

 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   protocolo        246 non-null    object
 1   data_abertura    246 non-null    object
 2   solicitante      246 non-null    object
 3   setor            236 non-null    object
 4   categoria        246 non-null    object
 5   assunto          246 non-null    object
 6   prioridade       232 non-null    object
 7   status           246 non-null    object
 8   data_fechamento  135 non-null    object
 9   avaliacao        124 non-null    float64
 10  unidade          246 non-null    object
dtypes: float64(1), object(10)
memory usage: 21.3+ KB

In [14]:
df.isna().sum()

,0
protocolo,0
data_abertura,0
solicitante,0
setor,10
categoria,0
assunto,0
prioridade,14
status,0
data_fechamento,111
avaliacao,122


resultado:
0
protocolo	0
data_abertura	0
solicitante	0
setor	10
categoria	0
assunto	0
prioridade	14
status	0
data_fechamento	111
avaliacao	122
unidade	0

In [18]:
quant = df.duplicated().sum()
print(quant)

6


resultado: 6

In [19]:
df["categoria"].nunique()

15

resultado 15

In [20]:
df["prioridade"].nunique()

8

resultado: 8

In [21]:
df["status"].nunique()

7

resultado: 7

In [23]:
contagem = df['unidade'].value_counts()
print(contagem)

unidade
Parambu       123
Solonópole    123
Name: count, dtype: int64


resultado:
unidade
Parambu       123
Solonópole    123
Name: count, dtype: int64